### DISCUSSION TITLE
Spotify messed up order

### DISCUSSION THREAD

**Discussion by Vasmares** [BimmerGestalt/AAIdrive#497] [Q&A] [answered]
Hi,

I dont know if this has something to do with your implementation of the spotify API or maybe its just spotify being stupid, I wanted to address this anyway:

When selecting / starting a new self created playlist the sorting is all messed up.
Especially with self created playlists. It does not start at track 01, but justr any other.

Funny enough, the sorting on spotify desktop and the android app is correct...
Any Ideas ?

See Screenshots
Car (wrong)
![photo_2022-01-11_18-59-13](https://user-images.githubusercontent.com/10863304/148997351-5761ef2a-8cc4-4f66-8f2b-4d6799aad57f.jpg)
![photo_2022-01-11_19-00-17](https://user-images.githubusercontent.com/10863304/148997356-1db6ed9a-f69f-4fb7-804d-678fb125406f.jpg)

Phone (correct)
![photo_2022-01-11_19-00-13](https://user-images.githubusercontent.com/10863304/148997364-4d2d4678-3c37-4d9f-96e7-c9d0be660205.jpg)
.

**Comment by drm87** [comment]
Is the shuffle mode on?

**Reply by Vasmares** [reply]
Not in the Spotify App.
Is there another option in the spotify Car UI as well ?

**Reply by Vasmares** [reply]
I have checked the options in the car. no, the shuffle mode is not on.

**Comment by Vasmares** [comment, marked as answer]
Update:
This is not caused by this app...
Spoify is just robbish at sorting.

It does not care for well maintained mp3 track information....
On custom playlists it sorts by the date the track has been added to the playlist....

So you need to order the tracks correctly before adding them to a playlist.
Then all is well. -ish

**Reply by hufman** [reply]
I'm glad you figured it out! Thank you for sharing what you found.
